stats/utils.q

```diff
@@ -0,0 +1,193 @@
+// @param endog {float[]} The endogenous variable
+// @param param {number[][]|number[]} A parameter to compare the length of
+// @param paramName {string} The name of the parameter
+// @returns {::;err} Return an error if they aren't equal
+stats.i.checkLen:{[endog;param;paramName]
+  if[not count[endog]=count param;
+    '"The length of the endog variable and ",paramName," must be equal"
+    ]
+  }
+
+// @private
+// @kind function
+// @category statsUtility
+// @desc Calculate descriptive stats for an OLS regression
+// @param coef {float[]} The coefficients for each predictor variable
+// @param endog {float[]} The endogenous variable
+// @param exog {float[][]} Values that predict the endog variable
+// @param trend {boolean} Whether a trend is added to the model
+// @returns {dictionary[]} The descriptive statistics
+stats.i.OLSstats:{[coef;endog;exog;trend]
+  n:count endog;
+  p:count[coef]-trend;
+  statsDict:stats.i.OLScalcs[coef;endog;exog;n;p];
+  variables:stats.i.coefStats[coef;endog;exog;trend;n;p];
+  `coef`variables`statsDict!(coef;variables;statsDict)
+  }
+
+// @private
+// @kind function
+// @category statsUtility
+// @desc Calculate descriptive stats for an OLS regression
+// @param coef {float[]} The coefficients for each predictor variable
+// @param endog {float[]} The endogenous variable
+// @param exog {float[][]} Values that predict the endog variable
+// @param n {long} The number of endog variables
+// @param p {long} Number of coefs not including trend value
+// @returns {dictionary[]} The descriptive statistics
+stats.i.OLScalcs:{[coef;endog;exog;n;p]
+  coefDict:enlist[`coef]!enlist coef;
+  modelInfo:enlist[`modelInfo]!enlist coefDict;
+  predicted:stats.OLS.predict[modelInfo;exog];
+  mseCalc:mse[predicted;endog];
+  r2:r2Score[predicted;endog];
+  r2Adj:r2AdjScore[predicted;endog;p];
+  // Calculate degrees of freedom
+  dfTotal:n-1;
+  dfModel:p;
+  dfResidual:dfTotal-dfModel;
+  // Sum of squares
+  SSTotal:sse[endog;avg endog];
+  SSModel:sse[predicted;first avg predicted];
+  SSResidual:sse[predicted;endog];
+  // Regression mean squares are the sum squares%degrees of freedom
+  MSTotal:SSTotal%dfTotal; 
+  MSModel:SSModel%dfModel;
+  MSResidual:SSResidual%dfResidual;
+  fStat:MSModel%MSResidual;
+  logLike:stats.i.logLiklihood[SSResidual;n];
+  rseCalc:rse[predicted;endog;dfResidual];
+  pValue:2*1-pyStats[`:t][`:cdf;<][fStat;p;dfResidual];
+  dictKeys:`dfTotal`dfModel`dfResidual`sumSquares`meanSquares,
+    `fStat`r2`r2Adj`mse`rse`pValue`logLike;
+  dictVals:(dfTotal;dfModel;dfResidual;SSResidual;MSResidual;fStat;r2;
+    r2Adj;mseCalc;rseCalc;pValue;logLike);
+  dictKeys!dictVals
+  }
+
+// @private
+// @kind function
+// @category statsUtility
+// @desc Calculate the logliklihood of the residuals
+// @param SSResiduals {float} Sum of squares of the residual
+// @param n {long} The number of endog variables
+// @returns {float[]} The loglikelihood value
+stats.i.logLiklihood:{[SSResidual;n]
```

Comment:
This should be `logLikelihood`, the change will need to be made in the description and within the OLScalcs function